pkg/controller/backup_sync_controller.go

```diff
@@ -107,4 +120,32 @@ func (c *backupSyncController) run() {
 			}
 		}
 	}
+
+	err = deleteUnused(c, cloudBackups)
+	if err != nil {
+		c.logger.WithError(errors.WithStack(err)).Error("Error deleting unused backup from Kubernetes")
+	}
+}
+
+// deleteUnused deletes a backup from Kubernetes if it does not exist in object storage
+func deleteUnused(c *backupSyncController, cloudBackups map[string]*api.Backup) error {
+	// Backups from Kubernetes
+	backups, err := c.backupLister.Backups(c.namespace).List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	if len(backups) == 0 {
+		return nil
+	}
+
+	// For each backup item in Kubernetes, verify if has a corresponding backup in the cloud object storage. If not, delete it.
+	for _, backup := range backups {
+		if _, ok := cloudBackups[backup.Name]; !ok {
+			c.client.Backups(backup.Namespace).Delete(backup.Name, &metav1.DeleteOptions{})
+			if err != nil {
+				return err
```

Comment:
do you want to break on the first error here? or should you delete as many as you can, and return a slice of errors encountered at the end?

Reply:
I wasn't clear if there could be multiples but this answers it! Also, thanks for the other comments with corresponding explanations.